src/bp-document/bp-document-functions.php

```diff
@@ -3761,3 +3449,1309 @@ function bb_document_user_can_upload( $user_id = 0, $group_id = 0 ) {
+				add_image_size( sanitize_key( $name ), $image_size['width'], $image_size['height'] );
+			}
+		}
+	}
+}
+
+/**
+ * Regenerate media attachment thumbnails
+ *
+ * @param int $attachment_id Attachment ID.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_document_regenerate_attachment_thumbnails( $attachment_id ) {
+
+	// Add upload filters.
+	bb_document_add_upload_filters();
+
+	// Register image sizes.
+	bb_document_register_image_sizes();
+
+	// Regenerate attachment thumbnails.
+	bp_core_regenerate_attachment_thumbnails( $attachment_id );
+
+	// Remove upload filters.
+	bb_document_remove_upload_filters();
+
+	// Deregister image sizes.
+	bb_document_deregister_image_sizes();
+}
+
+/**
+ * Add document upload filters.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_document_add_upload_filters() {
+	add_filter( 'upload_dir', 'bp_document_upload_dir' );
+	add_filter( 'intermediate_image_sizes_advanced', 'bb_document_remove_default_image_sizes' );
+	add_filter( 'upload_mimes', 'bp_document_allowed_mimes', 9, 1 );
+	add_filter( 'big_image_size_threshold', '__return_false' );
+}
+
+/**
+ * Remove document upload filters.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_document_remove_upload_filters() {
+	remove_filter( 'upload_dir', 'bp_document_upload_dir' );
+	remove_filter( 'intermediate_image_sizes_advanced', 'bb_document_remove_default_image_sizes' );
+	remove_filter( 'upload_mimes', 'bp_document_allowed_mimes' );
+	remove_filter( 'big_image_size_threshold', '__return_false' );
+}
+
+/**
+ * This will remove the default image sizes registered.
+ *
+ * @param array $sizes Image sizes registered.
+ *
+ * @return array Empty array.
+ * @since BuddyBoss 1.5.7
+ */
+function bb_document_remove_default_image_sizes( $sizes ) {
+	if ( isset( $sizes['bb-document-pdf-preview-activity-image'] ) && isset( $sizes['bb-document-pdf-image-popup-image'] ) && isset( $sizes['bb-document-image-preview-activity-image'] ) ) {
+		return array(
+			'bb-document-pdf-preview-activity-image'   => $sizes['bb-document-pdf-preview-activity-image'],
+			'bb-document-pdf-image-popup-image'        => $sizes['bb-document-pdf-image-popup-image'],
+			'bb-document-image-preview-activity-image' => $sizes['bb-document-image-preview-activity-image'],
+		);
+	}
+
+	return array();
+}
+
+/**
+ * Deregister document image sizes.
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_document_deregister_image_sizes() {
+	$image_sizes = bb_document_get_image_sizes();
+
+	if ( ! empty( $image_sizes ) ) {
+		foreach ( $image_sizes as $name => $image_size ) {
+			remove_image_size( sanitize_key( $name ) );
+		}
+	}
+}
+
+/**
+ * Add the fallback image sizes for PDF.
+ *
+ * @param $fallback_sizes
+ * @param $metadata
+ *
+ * @return mixed|void
+ *
+ * @since BuddyBoss 1.7.0
+ */
+function bb_document_fallback_intermediate_image_sizes( $fallback_sizes, $metadata ) {
```

Comment:
@chetansatasiya Add description in doc block params.

Reply:
done